### PR TITLE
refactor: replace OpenStruct with Data for performance and readability improvement

### PR DESCRIPTION
**Pull request by Yash-Singh-Pathania**
# Description

This PR refactors the codebase by replacing OpenStruct with Ordered Options for improved performance and immutability. The change ensures that Data is used to represent immutable objects, which provides better memory usage and execution speed compared to OpenStruct. No additional dependencies are required for this change.

Fixes # (issue)
[#3245](https://github.com/avo-hq/avo/issues/3245)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshots & recording

N/A (No UI changes involved)

## Manual review steps

1. Verify that the `accounts` method returns the correct objects using the new `Data` class.
2. Ensure that `perform_request` correctly handles the refactored `Response` class in test mode.

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: docs.avohq.io
Comment (adrianthedev) [inline docs.avohq.io:1]: I dont' think this should be here.
This is why the CI is failing.

**Comment by Yash-Singh-Pathania** [comment]
@adrianthedev  can you please  review this , I think its done sorry for the delay I just got started with my masters 
thanks a ton.

**Comment by Yash-Singh-Pathania** [comment]
@Paul-Bob  would you be reviewing this ?

**Review by Paul-Bob — CHANGES_REQUESTED** [review]
It feels unnatural to create a new class and override the initializer; it seems like we're not using the right tool for the job.

I came across this article discussing alternatives to `OpenStruct`:
https://dev.to/lorankloeze/alternative-for-rubys-openstruct-i72

I think we should consider using a more appropriate tool, such as `Struct` or `Data`.

`Struct` is significantly faster than `OpenStruct` (~80 times faster), although I’m unsure how it compares to `Data`. Let’s either research or run a quick benchmark, similar to the one in the article, to determine which option `Struct` or `Data` is faster, and proceed with that.

**Comment by Yash-Singh-Pathania** [comment]
@Paul-Bob  let me go through this quickly and run a quick benchmark for the same.

File: spec/dummy/app/models/user.rb [hunk 1]
```diff
@@ -19,7 +19,10 @@
 #  active                 :boolean          default(TRUE)
 #  slug                   :string
 #
+
```
Comment (Paul-Bob) [inline spec/dummy/app/models/user.rb:22]:
```suggestion
```

**Review by Paul-Bob — APPROVED** [review]
Thank you for working on this @Yash-Singh-Pathania 

I've completed the PR by using `Struct` instead of previous approaches.

**Pull request merged**